### PR TITLE
GraphQL.Server - Authentication & authorization tests (PR 12)

### PR DESCRIPTION
**Pull request by pekkah**
## Summary
- Add comprehensive test coverage for authentication and authorization patterns in GraphQL.Server
- Test integration with ASP.NET Core authentication/authorization through HttpContext.User
- Cover various auth middleware patterns that can be used in GraphQL request pipelines
- Validate security scenarios and edge cases for protecting GraphQL endpoints

## Test Coverage
- **User Authentication**: Test accessing authenticated vs unauthenticated users from HttpContext
- **Role-based Authorization**: Test middleware that restricts access based on user roles
- **Claims-based Authorization**: Test validation of specific claims and claim values
- **JWT Token Validation**: Test bearer token authentication middleware patterns
- **API Key Authentication**: Test API key-based authentication scenarios
- **Permission-based Authorization**: Test fine-grained permission systems
- **Rate Limiting**: Test per-user rate limiting middleware
- **Complex Authorization**: Test multi-factor authorization with roles, claims, and permissions
- **Error Handling**: Test proper error responses for unauthorized/unauthenticated requests

## Implementation Notes
- Uses mock authentication middleware implementations to demonstrate patterns
- Tests access to HttpContext.User and Claims from GraphQL request context
- Validates proper error response generation for security violations
- Covers both simple and complex authorization scenarios
- All middleware follow the IGraphQLRequestMiddleware pattern

## Security Scenarios Covered
- Unauthenticated user access attempts
- Missing required roles or claims
- Invalid JWT tokens and API keys
- Rate limiting enforcement
- Multi-layered authorization checks
- Proper error messages without information leakage

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: tests/GraphQL.Server.Tests/AuthenticationAuthorizationFacts.cs [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+            }
+
+            await next(context);
+        }
+    }
+
+    private class UserRateLimitMiddleware : IGraphQLRequestMiddleware
+    {
+        private readonly int _maxRequestsPerMinute;
+        private readonly Dictionary<string, List<DateTime>> _userRequests = new();
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Server.Tests/AuthenticationAuthorizationFacts.cs]: ✅ Fixed: Implemented thread-safe UserRateLimitMiddleware using ConcurrentDictionary and lock synchronization. Used GetOrAdd() for safe initialization and synchronized access to List<DateTime> operations to prevent race conditions in concurrent server environments.

File: tests/GraphQL.Server.Tests/AuthenticationAuthorizationFacts.cs [hunk 2]
```diff
@@ -0,0 +1,836 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
```
[the remark this answers is not in the document]
Comment (pekkah) [reply tests/GraphQL.Server.Tests/AuthenticationAuthorizationFacts.cs]: ✅ Fixed: Removed unused using directives (Microsoft.Extensions.DependencyInjection and Microsoft.Extensions.Primitives) to improve code clarity and reduce namespace pollution.
